src/server/graph-generation/interaction/index.js

```diff
@@ -65,11 +67,50 @@ function parse(data, queryIds){
       const edgeMetadata = interactionMetadata(splitLine[6],splitLine[4]);
       addInteraction([splitLine[0], splitLine[2]], splitLine[1], edgeMetadata, network, nodeMap, nodeMetadata, queryIds);
     });
+
+    const cy = cytoscape({headless:true,container:undefined,elements:network});
+    const bc = cy.$().bc();
+    const nodes = cy.nodes();
+
+    if(nodes.length === 0)
+      return network;
+
+    //loop through the nodes, collected betweenness centrality values
+    let centralityVals = [];
+    let centralityMap = [];
+    nodes.forEach( ele => {
+      let bcVal = bc.betweenness(ele);
+      centralityVals.push(bcVal);
+      centralityMap.push([ele,bcVal]);
+    });
+
+    const max = Math.max(...centralityVals);
+    const min = Math.min(...centralityVals);
+
+    if(max !== min)
+      centralityMap.forEach( ele => {
+        ele[0].data('bcVal',normalizeValue(ele[1],max,min));
+      });
+    else
+      centralityMap.forEach( ele => {
+        ele[0].data('bcVal',ele[1]);
+      });
```

Comment:
Let's make the calculation more abstract.

(1) Do the calculation in a different function, addMetric().
(2) The code that uses parse() can call both parse() and then addMetric().
(3) Don't normalise the values yourself.  The algorithm can do it.  Use the normalised variant.
(4) Store the value under the node in a key called something general like `metric` in case we change the actual value used as the metric